onnx/optimizer/optimize.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "onnx/py_utils.h" // Keep this at the top to avoid the compilation warning.
+
+#include <list>
+
+#include "onnx/ir.h"
+#include "onnx/ir_pb_converter.h"
+#include "onnx/optimizer/passes/eliminate_nop_transpose.h"
+#include "onnx/optimizer/passes/fuse_consecutive_transposes.h"
+#include "onnx/optimizer/passes/fuse_transpose_into_gemm.h"
+#include "onnx/optimizer/passes/split.h"
+
+namespace onnx { namespace optimization {
+
+void PrepareOutput(const onnx::ModelProto& mp_in, onnx::ModelProto& mp_out);
+
+struct Optimizer {
+  std::map<std::string, std::unique_ptr<OptimizePass>> passes;
+
+  Optimizer() {
+    // Register the optimization passes to the optimizer.
+    std::unique_ptr<FuseConsecutiveTransposes> fct(new FuseConsecutiveTransposes());
+    passes[fct->name] = std::move(fct);
+    std::unique_ptr<EliminateNopTranspose> ent(new EliminateNopTranspose());
+    passes[ent->name] = std::move(ent);
+    std::unique_ptr<FuseTransposeIntoGemm> ftg(new FuseTransposeIntoGemm());
+    passes[ftg->name] = std::move(ftg);
+    std::unique_ptr<SplitInit> si(new SplitInit());
+    passes[si->name] = std::move(si);
+    std::unique_ptr<SplitPredict> sp(new SplitPredict());
+    passes[sp->name] = std::move(sp);
+  }
+
+  std::string optimize(const std::string& content, std::list<std::string>& names) {
```

Comment:
why "content"? should be either ModelProto or Graph following the pattern of optimizepass design below.

Reply:
Definitely in-memory proto directly, not serialized

For a follow up discussion - we might need to build a separate API to abstract out initializers. Serializing them in memory sometimes is prohibitively expensive for big models

Reply:
Actually this is API to bridge the python world and C++ world. Unless we really want to bind python and C++ ModelProto.